fvm/handler/computation.go

```diff
@@ -27,36 +29,38 @@ type SubComputationMeter interface {
 	// Commit()
 }
 
-// ComputationMeteringHandler handles computation metering on a transaction level
-type ComputationMeteringHandler interface {
-	ComputationMeter
-	StartSubMeter(limit uint64) SubComputationMeter
-}
-
 type computationMeter struct {
-	used  uint64
-	limit uint64
+	used        uint64
+	limit       uint64
+	intensities map[string]uint64
+	handler     *ComputationMeteringHandler
 }
 
 func (c *computationMeter) Limit() uint64 {
 	return c.limit
 }
 
-func (c *computationMeter) AddUsed(used uint64) error {
-	c.used += used
-	return nil
+func (c *computationMeter) AddUsed(intensity uint64, feature string) {
+	c.intensities[feature] += intensity
+	if feature != "function_or_loop_call" {
```

Comment:
what about statements?

Reply:
This is the label used for computation coming from cadence:

```
func (e *ScriptEnv) SetComputationUsed(used uint64) error {
	return e.computationHandler.AddUsed(used)
	e.computationHandler.AddUsed(used, "function_or_loop_call")
	return nil
}
```

Which includes everything we measured before (including statements).